docs/rules/require-homepage.md

```diff
@@ -0,0 +1,78 @@
+# require-homepage
+
+💼 This rule is enabled in the 📦 `recommended-publishable` config.
```

Comment:
Interesting.  We didn't really discuss this being part of the publishable config.  Seems reasonable but just want to double check, should it be?  What do you think @JoshuaKGoldberg?  (We'll need to mark this as breaking, if we include this change. - which is totally cool, but highlighting it as a change needed)

Reply:
Oh, actually, I took a second to think about this, and _don't_ think this should be in the publishable config.  There are different kinds of publishing that happen.  Some are public, which is where this would be good to require, but every company I've worked for in the last 15 years has also had an internal registry where packages can be published for internal use.  And it wouldn't make sense to require `homepage` in those cases.  So, let's not include it for now.

Reply:
> For this reason maybe we should set ignorePrivateDefault to true?

That wouldn't really work.  `private` in a package.json is different from an internally hosted corporate registry.  You still have to have `private: true` in order to publish to _any_ registry, whether it's npmjs.org or an internal registry.